Tools/CMake/AMReXCUDAOptions.cmake

```diff
@@ -0,0 +1,95 @@
+include_guard(GLOBAL)
+
+include(CMakeDependentOption)
+
+#
+# Define a macro to print active options
+#
+macro (cuda_print_option _var)
+   if (${_var})
+      message( STATUS "   ${_var}")
+   endif ()
+endmacro ()
+
+
+#
+#  CUDA-related options
+#
+message(STATUS "Enabled CUDA options:")
+
+set(AMReX_CUDA_ARCH_DEFAULT "Auto")
+if(DEFINED ENV{AMREX_CUDA_ARCH})
```

Comment:
@ax3l How should we handle `CMAKE_CUDA_ARCHITECTURES`?

Reply:
The usual logic in CLI and env vars on Unix is: the more specific the name, the higher the precedence.

So env var [CUDAARCH](https://cmake.org/cmake/help/v3.20/envvar/CUDAARCHS.html#envvar:CUDAARCHS) and according [CMAKE_CUDA_ARCHITECTURES](https://cmake.org/cmake/help/v3.20/variable/CMAKE_CUDA_ARCHITECTURES.html?highlight=cmake_cuda_architectures) have lower precedence than the specific `AMREX_CUDA_ARCH` if multiple are present.

If `AMREX_CUDA_ARCH` is not set, we honor `CUDAARCH`/`CMAKE_CUDA_ARCHITECTURES` and if they are also not set we do auto-detection/defaults.

Reply:
It should be taken care of now